components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2123,33 +2123,51 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    protected defaultSpendingLimit = 100;
-    async subscribeTeamToStripe(ctx: TraceContext, teamId: string, setupIntentId: string): Promise<void> {
-        this.checkAndBlockUser("subscribeUserToStripe");
-        const team = await this.guardTeamOperation(teamId, "update");
-        await this.ensureStripeApiIsAllowed({ team });
+    async subscribeToStripe(ctx: TraceContext, attributionId: string, setupIntentId: string): Promise<void> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
+
+        const user = this.checkAndBlockUser("subscribeToStripe");
+
+        let customer: Stripe.Customer | undefined;
         try {
-            let customer = await this.stripeService.findCustomerByTeamId(team!.id);
+            if (attrId.kind === "team") {
+                const team = await this.guardTeamOperation(attrId.teamId, "update");
+                await this.ensureStripeApiIsAllowed({ team });
+                customer = await this.stripeService.findCustomerByTeamId(team!.id);
+            } else {
+                await this.ensureStripeApiIsAllowed({ user });
+                customer = await this.stripeService.findCustomerByUserId(user.id);
+            }
             if (!customer) {
-                throw new Error(`No Stripe customer profile for team '${team.id}'`);
+                throw new Error(`No Stripe customer profile for '${attributionId}'`);
```

Comment:
```suggestion
                throw new Error(`Could not find Stripe customer profile for '${attributionId}'. Please reach out to our support team.`);
```
Just feels we can be friendlier when this does blow up.